db/migrate/20190813134515_add_devise_to_users.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+class AddDeviseToUsers < ActiveRecord::Migration[5.2]
+  def self.up
+    ## Database authenticatable
+    # add_column string :email, null: false, default: ""
+    # add_column :users, :encrypted_password, :string, null: false, default: ""
+    rename_column :users, :password, :encrypted_password
+
+    ## Recoverable
+    add_column :users, :reset_password_token, :string
+    add_column :users, :reset_password_sent_at, :datetime
+
+    ## Trackable
+    add_column :users, :sign_in_count, :integer, default: 0
+    add_column :users, :current_sign_in_at, :datetime
+    add_column :users, :last_sign_in_at, :datetime
+    add_column :users, :current_sign_in_ip, :inet
+    add_column :users, :last_sign_in_ip, :inet
+
+    ## Confirmable
+    add_column :users, :confirmation_token, :string
+    add_column :users, :confirmed_at,:datetime
+    add_column :users, :confirmation_sent_at, :datetime
+    #add_column :users, :unconfirmed_email string# Only if using reconfirmable
+
+    ## Lockable
+    add_column :users, :failed_attempts, :integer, default: 0 # Only if lock strategy is :failed_attempts
+    add_column :users, :unlock_token, :string # Only if unlock strategy is :email or :both
+    add_column :users, :locked_at, :datetime
+
+    change_column :users, :nickname, :string, default: '', null: true
```

Comment:
Change 32-35 lines to `null: false`

Reply:
But remember at first rollback, than change, than migrate again

Reply:
Why?